.github/workflows/build.yml

```diff
@@ -22,11 +22,8 @@ jobs:
           go-version: '>=1.18.0'
       - run: wget https://github.com/google/flatbuffers/releases/download/v22.9.29/Linux.flatc.binary.g++-10.zip
       - run: unzip Linux.flatc.binary.g++-10.zip
-      - run: make install-check-tools
-      - run: make check-ltag
-      - run: make check-dco
-      - run: make check-lint
-      - run: PATH=$PATH:$(pwd) make check-flatc
+      - run: ./scripts/install-check-tools.sh
+      - run: PATH=$PATH:$(pwd) ./scripts/check.sh
```

Comment:
We had decided to keep all check tests separate to help devs debug which check failed exactly. Can we do the same here, by having all check tests in separate files?

Reply:
SGTM. I'm also planning to just remove golangci-lint related stuff. I didn't see other repos have rules/scripts for it and installing/running it is straightforward (the script will be just 1 line anyway). And we should use `golangci/golangci-lint-action` in our CI for it. Any concern with this?

Reply:
SGTM. But we need to make sure we also lint inside the `cmd` directory; since it's a separate package.

Reply:
`golangci-lint run` without args uses `./...` and runs in all (sub)directories. https://golangci-lint.run/usage/quick-start/

Will also test in the action run.

Reply:
changed back to use script for golangci-lint. turns out it doesn't support multi-mod repo and will just ignore those subdirectories with own `go.mod`: https://github.com/golangci/golangci-lint/issues/828